### ISSUE TITLE
about inference

### ISSUE THREAD

**Issue by finger-monkey** [yassouali/pytorch_segmentation#73]
hello,I want to use the trained model 'PSPnet.pth' you've provide , to inference directly on  my picture。But your code is related to the training phase and involves some data set information,  I just want to use your model as a tool directly . How can I avoid all this? help me, please.

def main():
    args = parse_arguments()
    config = json.load(open(args.config))

    # Dataset used for training the model
    dataset_type = config['train_loader']['type']
    assert dataset_type in ['VOC', 'COCO', 'CityScapes', 'ADE20K']
    if dataset_type == 'CityScapes': 
        scales = [0.75, 1.0, 1.25, 1.5, 1.75, 2.0, 2.25] 
    else:
        scales = [0.75, 1.0, 1.25, 1.5, 1.75, 2.0]
    scales = [0.75, 1.0, 1.25, 1.5, 1.75, 2.0]
    loader = getattr(dataloaders, config['train_loader']['type'])(**config['train_loader']['args'])
    to_tensor = transforms.ToTensor()
    normalize = transforms.Normalize(loader.MEAN, loader.STD)
    num_classes = loader.dataset.num_classes
    palette = loader.dataset.palette

    # Model
    model = getattr(models, config['arch']['type'])
    device = torch.device( 'cpu')

    checkpoint = torch.load('PSPnet.pth', map_location=device)

    if isinstance(checkpoint, dict) and 'state_dict' in checkpoint.keys():
        checkpoint = checkpoint['state_dict']

    if 'module' in list(checkpoint.keys())[0] and not isinstance(model, torch.nn.DataParallel):
        model = torch.nn.DataParallel(model)
    model.load_state_dict(checkpoint)
    model.to(device)
    model.eval()

**Comment by yassouali** [comment]
Hi,

When doing inference, the config file is required to just create the same model used during training, in your case, if you want to use the provided PSPNet, you can simply load the config file in the .pth file.

try adding the following after  `checkpoint = torch.load ...`

```python
if  'config' in checkpoint.keys():
    config = checkpoint['config']
```
